.github/workflows/release_snapshot.yml

```diff
@@ -75,6 +81,15 @@ jobs:
           python release_creation/main.py \
            --input-version=${{ inputs.version_number }} \
            --operation=update
+  linux-test:
+    needs: [linux, linux-create]
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.9", "3.10"]
```

Comment:
I think I would have a `update_snapshots` job that has a matrix of

``` 
   runs-on: ${{ matrix.os }}

    strategy:
      matrix:
        python-version: ["3.7", "3.9", "3.10", "3.11"]
        os: [ubuntu-latest, macos-latest]
        include:
          - python-version: 3.8
            os: macos-latest
```

Then a `test_snapshots` job with

``` 
   runs-on: ${{ matrix.os }}

    strategy:
      matrix:
        python-version: ["3.7", "3.9", "3.10", "3.11"]
        os: [ubuntu-latest, macos-latest, macos-11]
        include:
          - python-version: 3.8
            os: macos-latest
          - python-version: 3.8
            os: macos-11
```
possibly just add in python 3.8 here instead of when you initially create the snapshot unless you test it after the initial creation as a bit of a sanity check things will work before creating the rest.

There are a few "do this if linux, that is mac" that you'd have to code.  But I think it makes the flow a bit more clear.

Reply:
great suggestion, thanks!